### PR TITLE
sys-kernel: move dracut build to coreos-kernel

### PR DESCRIPTION
**Pull request by crawford**
Rather than building the initramfs in the postinst of bootengine, this
moves it to the setup of coreos-kernel. This has the advantage of
failing a build if dracut fails and ensuring that the latest initramfs
makes it into every kernel build.

### CODE REVIEW

**Comment by mischief** [comment]
please rebase after #1658

File: sys-kernel/coreos-kernel/coreos-kernel-4.2.2-r10.ebuild [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	sys-apps/sed
+	sys-apps/shadow
+	sys-apps/systemd
+	sys-apps/util-linux
+	sys-fs/btrfs-progs
+	sys-fs/mdadm
+	sys-kernel/dracut
+	virtual/udev
+	"
+RDEPEND="${DEPEND}"
```
Comment (marineam) [inline sys-kernel/coreos-kernel/coreos-kernel-4.2.2-r10.ebuild]: This is going to pull ignition into the root filesystem of prod images. Just use DEPEND instead.

File: sys-kernel/coreos-kernel/coreos-kernel-4.2.2-r10.ebuild [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+# We are bad, we want to get around the sandbox.  So do the creation of the
+# cpio image in pkg_setup() where we are free to mount filesystems, chroot,
+# and other fun stuff.
+pkg_setup() {
```
Comment (marineam) [inline sys-kernel/coreos-kernel/coreos-kernel-4.2.2-r10.ebuild]: Check for `"${MERGE_TYPE}" == binary` and return. No need to run this when not building the source. Also important to not run this for binary packages when using DEPEND but not RDEPEND as noted above.

**Comment by marineam** [comment]
Why not put this in the eclass?

**Comment by crawford** [comment]
Moved to the eclass and checking against MERGE_TYPE.

File: sys-kernel/coreos-kernel/coreos-kernel-4.3.0.ebuild [hunk 3]
```diff
@@ -7,3 +7,21 @@ inherit coreos-kernel
 
 DESCRIPTION="CoreOS Linux kernel"
 KEYWORDS="amd64 arm64"
+
+DEPEND="
```
Comment (marineam) [inline sys-kernel/coreos-kernel/coreos-kernel-4.3.0.ebuild]: These should also be in the eclass.

Comment (marineam) [inline sys-kernel/coreos-kernel/coreos-kernel-4.3.0.ebuild]: Also, if any of them have sub-slots you can add := to trigger the rebuild, similar to the bootengine dependency in the eclass.

Comment (marineam) [inline sys-kernel/coreos-kernel/coreos-kernel-4.3.0.ebuild]: nm, doesn't look like any do, but you could add that to ignition.

File: sys-kernel/coreos-kernel/coreos-kernel-4.3.0.ebuild [hunk 4]
```diff
@@ -7,3 +7,21 @@ inherit coreos-kernel
...
+	sys-apps/coreutils
+	sys-apps/findutils
+	sys-apps/grep
+	sys-apps/ignition
+	sys-apps/less
+	sys-apps/sed
+	sys-apps/shadow
+	sys-apps/systemd
+	sys-apps/util-linux
+	sys-fs/btrfs-progs
```
Comment (mischief) [inline sys-kernel/coreos-kernel/coreos-kernel-4.3.0.ebuild]: do you need sys-fs/xfsprogs here as well...?

Comment (marineam) [inline sys-kernel/coreos-kernel/coreos-kernel-4.3.0.ebuild]: yeah, that should be there too. e2fs also

File: sys-kernel/coreos-kernel/coreos-kernel-4.3.0.ebuild (hunk 3 above)
Comment (crawford) [inline sys-kernel/coreos-kernel/coreos-kernel-4.3.0.ebuild]: Added

File: sys-kernel/coreos-kernel/coreos-kernel-4.3.0.ebuild (hunk 4 above)
Comment (crawford) [inline sys-kernel/coreos-kernel/coreos-kernel-4.3.0.ebuild]: Added

File: eclass/coreos-kernel.eclass [hunk 5]
```diff
@@ -20,8 +20,26 @@ SLOT="0/${PVR}"
...
+	sys-apps/coreutils
+	sys-apps/findutils
+	sys-apps/grep
+	sys-apps/ignition:=
+	sys-apps/less
+	sys-apps/sed
+	sys-apps/shadow
+	sys-apps/systemd
+	sys-apps/util-linux
+	sys-fs/btrfs-progs
```
Comment (mischief) [inline eclass/coreos-kernel.eclass]: i suppose btrfs-progs would pull in sys-fs/e2fsprogs, but it might be better to be explicit about it

Comment (marineam) [inline eclass/coreos-kernel.eclass]: yeah, clearly the deps work as-is but explicit helps. :)

**Comment by marineam** [comment]
LGTM

**Comment by crawford** [comment]
Hmm, I bumped Ignition and ran `./build_packages`. It rebuilt Ignition, but not the kernel.

**Comment by marineam** [comment]
Hm... maybe it thinks it is ok because it isn't a runtime (RDEPEND) then. Bah...

**Comment by marineam** [comment]
but I swear it worked for bootengine which has always been DEPEND not RDEPEND

**Comment by marineam** [comment]
when you did this had both the kernel and ignition been built since you added the subslot and dependency?

**Comment by crawford** [comment]
No, neither had been built. Should the kernel rev be bumped to pick up this change?

**Comment by marineam** [comment]
the installed kernel will need to be built with the := operator in place in order for a change in subslot to trigger it. as far as portage was concerned the deps on the installed version were still valid.

**Comment by marineam** [comment]
so a rev-bump isn't strictly required since in the end on-disk the same thing gets installed but a rebuild is required before := starts working. a rev-bump will force that.

**Comment by crawford** [comment]
That did it. I'll include the rev bump.

**Comment by marineam** [comment]
most of the time the in-ebuild dependencies will amend the recorded dependencies of installed packages, this allows for dependency fixes to be made without requiring users pointlessly compile. apparently the behavior of the := operator doesn't apply though. could be any number of weird reasons why.

**Comment by marineam** [comment]
LGTM

**Pull request merged**